src/sentry/static/sentry/app/__mocks__/api.jsx

```diff
@@ -32,14 +32,19 @@ class Client {
         headers: response.headers || {},
       },
       mock,
+      predicate,
     ]);
 
     return mock;
   }
 
   static findMockResponse(url, options) {
-    return Client.mockResponses.find(([response]) => {
-      return url === response.url && (options.method || 'GET') === response.method;
+    return Client.mockResponses.find(([response, mock, predicate]) => {
+      const matchesURL = url === response.url;
+      const matchesMethod = (options.method || 'GET') === response.method;
+      const matchesPredicate = predicate(url, options);
+
+      return matchesURL && matchesMethod && matchesPredicate;
```

Comment:
i'm adding a predicate function here to further filter mocked responses based on `options`, rather than just the `url` alone.